HomeWork1/HW1_NazarBozhik

```diff
@@ -0,0 +1,54 @@
+#include <cmath>
+#include <iostream>
+
+int main() {
+  std::cout << "Please enter three numbers in range [2..255]\n";
+  const int cF = 4;
+  const int cT = 2;
+  float a, b, c;
+  std::cout << "Enter a =";
+  std::cin >> a;
+  while (a <= 2 || a >= 255) {
+    std::cout << "Enter a valid numder[2..255]\n";
+    std::cin.clear();
+    std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+    std::cout << "Enter a =";
+    std::cin >> a;
+  }
+  std::cout << "Enter b =";
+  std::cin >> b;
+  while (b <= 2 || b >= 255) {
+    std::cout << "Enter a valid numder[2..255]\n";
+    std::cin.clear();
+    std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+    std::cout << "Enter b =";
+    std::cin >> b;
+  }
+  std::cout << "Enter c =";
+  std::cin >> c;
+  while (c <= 2 || c >= 255) {
+    std::cout << "Enter c valid numder[2..255]\n";
+    std::cin.clear();
+    std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+    std::cout << "Enter c =";
+    std::cin >> c;
+  }
+  auto D = pow(b, 2) - cF * a * c;
+
+  // std::cout << sqrt(D); //проверял роботу корня
+  if (D > 0) {
+    std::cout << "We will have x1 and x2, because our D > 0\n";
+    const auto xf = -b - sqrt(D) / cT * a;
+    std::cout << "x1 =" << xf << "\n";
+    const auto xd = -b + sqrt(D) / cT * a;
+    std::cout << "x2 =" << xd << "\n";
+  } else if (D == 0) {
+    std::cout << "We will have just one x, because our D = 0\n";
+    const auto xs = -b / cT * a;
+    std::cout << "x =" << xs;
+  } else {
+    std::cout << "D < 0, so we don't have variants";
+  }
```

Comment:
как бы все ок, но я не люблю понятия `if else` он немного запутывает чтения вашего кода, так как сложно отследить ваши ветки.
```
if (D < 0) {
  std::cout << "D < 0, so we don't have variants";
  return 1;
}

if (D == 0) {
  std::cout << "We will have just one x, because our D = 0\n";
  const auto xs = -b / cT * a;
  return 0;
}
std::cout << "We will have x1 and x2, because our D > 0\n";
const auto xf = -b - sqrt(D) / cT * a;
std::cout << "x1 =" << xf << "\n";
const auto xd = -b + sqrt(D) / cT * a;
std::cout << "x2 =" << xd << "\n";
return 0;
```